### PR TITLE
groups: fix groups reconciliation, misc cleanup

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/460
- Followup to: https://github.com/kubernetes/k8s.io/pull/2401
- Possibly supercedes: https://github.com/kubernetes/k8s.io/pull/2413

This is https://github.com/kubernetes/k8s.io/pull/2413 but with all the extra stuff I was doing along the way before I decided... wait... just the few-line fix first.

Split into distinct commit explaining what I was doing, but I'm happy to squash after review

### CODE REVIEW

File: groups/reconcile.go [hunk 1]
```diff
@@ -246,40 +246,48 @@ func main() {
 	}
 }
 
-func readConfig(configFilePath string, confirmChanges bool) error {
-	if *verbose {
-		log.Printf("reading config file %s", configFilePath)
-	}
+func (c *Config) Load(configFilePath string, confirmChanges bool) error {
+	log.Printf("reading config file: %s", configFilePath)
```
Comment (nikhita) [inline groups/reconcile.go:250]: nit: these logs are enabled at other places only if `verbose` is true. Is there a reason why we remove that check here?

**Review by nikhita — APPROVED** [review]
One nit but we can probably get to that in a follow-up.

/lgtm
/label tide/merge-method-squash

/hold
please cancel the hold when ready

File: groups/reconcile.go (hunk 1 above)
Comment (spiffxp) [reply groups/reconcile.go:250]: I don't know that I accomplished this cleanly, but what I tried to do was have top-level config type stuff output by default, and leave verbose for individual operations.  The reason being that I, personally, find a lot more comfort in getting a sense of what this is going to do and where it's reading from before it does things... but then I only want to know when it's making changes (vs. walking though things)

I realize in running the tests this makes things a little noisy for non-CLI-invocation use-cases.  But the tests are it, for now, and there aren't many.

Were someone to keep going on refactoring this for testability, I would suggest using an actual logging package with verbosity levels, such that the really granular stuff could be dropped to a higher verbosity level (debug or v=2 or something), and configuration of that logging to be done by the main entrypoint vs. a much quieter default used by tests.

**Pull request merged**